test/e2e_node/hugepages_test.go

```diff
@@ -178,12 +179,57 @@ func runHugePagesTests(f *framework.Framework) {
 		err := e2epod.WaitForPodSuccessInNamespace(f.ClientSet, verifyPod.Name, f.Namespace.Name)
 		framework.ExpectNoError(err)
 	})
+
+	ginkgo.It("should add resources for new huge page sizes on kubelet restart", func() {
+		ginkgo.By("Stopping kubelet")
+		startKubelet := stopKubelet()
+		ginkgo.By(`Patching away support for hugepage resource "hugepages-2Mi"`)
+		patch := []byte(`[{"op": "remove", "path": "/status/capacity/hugepages-2Mi"}, {"op": "remove", "path": "/status/allocatable/hugepages-2Mi"}]`)
+		result := f.ClientSet.CoreV1().RESTClient().Patch(types.JSONPatchType).Resource("nodes").Name(framework.TestContext.NodeName).SubResource("status").Body(patch).Do(context.TODO())
+		framework.ExpectNoError(result.Error(), "while patching")
+
+		ginkgo.By("Starting kubelet again")
+		startKubelet()
+
+		ginkgo.By("verifying that the hugepages-2Mi resource is present")
+		gomega.Eventually(func() bool {
+			node, err := f.ClientSet.CoreV1().Nodes().Get(context.TODO(), framework.TestContext.NodeName, metav1.GetOptions{})
+			framework.ExpectNoError(err, "while getting node status")
+			_, isPresent := node.Status.Capacity["hugepages-2Mi"]
+			return isPresent
+		}, 30*time.Second, framework.Poll).Should(gomega.Equal(true))
+	})
 }
 
 // Serial because the test updates kubelet configuration.
 var _ = SIGDescribe("HugePages [Serial] [Feature:HugePages][NodeFeature:HugePages]", func() {
 	f := framework.NewDefaultFramework("hugepages-test")
 
+	ginkgo.It("should remove resources for huge page sizes no longer supported", func() {
```

Comment:
this is a rare test that we can exercise with hugepages in shared testing infrastructure as it does not depend on native support in the test environment.